components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java

```diff
@@ -97,35 +105,66 @@ private void syncSolaceApplicationClientId(ApplicationRegistrationEvent event) t
                     if (gatewayEnvironments.containsKey(deployment.getDeployment())) {
                         if (SolaceConstants.SOLACE_ENVIRONMENT.equalsIgnoreCase(gatewayEnvironments.get(deployment.
                                 getDeployment()).getProvider())) {
+                            APIProvider apiProvider = APIManagerFactory.getInstance().getAPIProvider(CarbonContext.
```

Comment:
can't we internally retrieve this ?

Reply:
We need an API object to be passed to create an application Solace broker.
That is retrieved from the subscriprtionEvent in subscription flow. In key generation flow, we don't have that object in the event and we need to access the DAO layer.

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/11173/commits/b4731c797a76dce959fc2727b1530e0660ef2b54